DataDog/datadog-agent
# Remove retry queue from forwarder

### What does this PR do?

This PR remove `retryQueue` from `domainForwarder`. It also deprecate the settings `forwarder_retry_queue_max_size`.

### Motivation

https://github.com/DataDog/datadog-agent/pull/6921#discussion_r538387632

### Additional Notes

Reviewing commit by commit is recommended.

### Describe your test plan

Checks on Agent Core side:
 - The memory retry queue is still working (When `forwarder_retry_queue_payloads_max_size > 0` and `forwarder_storage_max_size_in_bytes` = 0).
- Check the behavior and the warning message when
   - `forwarder_retry_queue_max_size` > 0  `forwarder_retry_queue_payloads_max_size` = 0
   - `forwarder_retry_queue_max_size` > 0  `forwarder_retry_queue_payloads_max_size` > 0
   - `forwarder_retry_queue_max_size` = 0  `forwarder_retry_queue_payloads_max_size` = 0
   - `forwarder_retry_queue_max_size` = 0  `forwarder_retry_queue_payloads_max_size` > 0

Check on processes side:
- Check the code https://github.com/DataDog/datadog-agent/pull/7079/commits/c645ce1dbc0c7b5d5b3c97bd6a892f07b3ac30e7#diff-993d2491e7efac8e13349ef7859c375a68e328aa5415289b699a6da0cc2637c3R218-R227
